src/index.js

```diff
@@ -233,23 +233,19 @@ function createLoadableComponent(loadFn, options) {
       clearTimeout(this._timeout);
     }
 
-    retry = () => {
-      if (!this._mounted) {
-        return;
-      }
-      this.setState({ error: null, loading: true, timedOut: false });
-      res = loadFn(opts.loader);
-      this._loadModule();
-    };
-
     render() {
       if (this.state.loading || this.state.error) {
+        // NOTE(kevinb): webapp has an ErrorBoundary which will catch and log any
+        // exceptions.  By throwing the error instead of passing it to the `loading`
+        // component we ensure that the error will be bubbled up without devs needing
+        // to explicitly check if there's an error in each `loading` component.
+        if (this.state.error) {
+          throw this.state.error;
+        }
```

Comment:
**suggestion**: I think this change goes too far. I think that the existing setup is really useful in some situations. For example, one may want to use a simple fallback UX within the component rather than creating a totally separate wrapper error boundary just to support that. There may also be times when retrying is useful.

I wonder if we could instead have this be an error boundary and then, if the child component throws during render, we catch that and, if we already have a `this.state.error`, throw that to our parent instead; otherwise rethrow the caught error?

If we can't do that, perhaps having a `throwOnError` prop (default: `true`) instead so that folks can choose to opt-out of the throwing behavior and use the existing path?

Reply:
I like having a `throwOnError` prop since there could be a situation where we catch an error thrown by the child, but `this.state.error` hasn't been set.  It would be nice to allow clients the option to handle errors in that situation.

Reply:
I think even if `this.state.error` is set, we'd want the option of handling it within the component rather than going straight to error boundary.

Reply:
The simplest change would be to add a `throwOnError` prop that is defaulted to `true` and then if it is false, do the old way. If it's `true`, do the new way. Anything else is hard to grok for folks, I think.

Reply:
Agreed.  I'm going to go with `throwOnError`. 🙂

Reply:
@somewhatabstract do we already have a use case that needs the ability to opt out?  Or would it make sense to ship what we have here and when the need arises for something different then we tweak?

Every option we add is another bit of mental load we are imposing on engineers.  So Im usually hesitant to add options unless there is a specific use case.

Reply:
It feels weird to delete existing code and then add it back. I'd rather make careful changes with meaningful defaults like I'm suggestion. I understand the argument of already done code is less work than other code, but that argument can be applied to the current state that we are looking to change - which is stable. What I'm proposing minimizes changes ensuring that there's less room for unforeseen errors.

As for if it's currently used, I doubt the `retry` function is. I have used the `error` check in one or two cases, though I don't think it is in use right now (the one I recall was around conditionally loading a library and handling if it wasn't available - and I think we removed support for that library).

Reply:
In addition, error boundaries don't fully work server-side. We have a hack for that. The existing solution supports SSR.

Reply:
> It feels weird to delete existing code and then add it back
I understand.  I think the balance is - add code back when we need it with contextual comments and test cases.  Or keep code that we don't currently need and probably won't need and yet we have to maintain it.  Imagine migrating to a new version of Loadable and then some years later someone has to figure out if the option we have added needs to be ported.

> In addition, error boundaries don't fully work server-side. We have a hack for that around specific boundaries. The existing solution supports SSR.
Oh that's a good point!  The issue here is tho that server side rendering is just going to get stuck showing a spinner, which is probably harder to catch than throwing an error...?

Anyways - I personally dont think is weird to remove code and add it back later when we have more information about the use case.   But if yall feel strongly about adding such option, Im happy too :)

Reply:
> The issue here is tho that server side rendering is just going to get stuck showing a spinner, which is probably harder to catch than throwing an error...?

Yes, it would depend on the situation of course, but if the rendered code didn't throw and showed something blank in light of an error, then we could use the `error` value to render something different on the server. Otherwise, if we just throw the error, we'll render our custom error boundary stuff (via the aforementioned hackery). I think keeping the existing solution gives us a very useful escape hatch when we need it.

Reply:
`throwOnError` is default to true which will cause any errors to be caught by webapp's `ErrorBoundary`.  If anyone sets it to `false` they'll need check `props.error`.  I'll include a warning in the docs about this.